### ISSUE TITLE
Wrong color priority in ruby files

### ISSUE THREAD

**Issue by dominikduda** [luochen1990/rainbow#70]
Hello,

Colors mess up in ruby files inside methods and classes.

![image](https://cloud.githubusercontent.com/assets/14922999/22830022/5c6e8d20-efa5-11e6-87fd-0403ad6dcb28.png)

That is pretty annyoing, because I want first level of parentheses to be text color.

In other file types it works correctly.

My config:
```viml
let g:rainbow_conf = {
    \   'ctermfgs': ['red', 'green', 'blue', 'black', 'orange', 'magenta'],
    \   'operators': '_,_',
    \   'parentheses': ['start=/(/ end=/)/ fold', 'start=/\[/ end=/\]/ fold', 'start=/{/ end=/}/ fold'],
    \}
```

Is that fixable?

**Comment by bartoszmaka** [comment]
Im experiencing this issue as well

**Comment by luochen1990** [comment]
I'm not sure what effect you are expecting, you want to fix the outmost color to blue, is that right ?

**Comment by dominikduda** [comment]
I'm expecting that all the parentheses will be colored as set in settings - same as in line 4, 5, and 6.

Yet inside methods, it breaks.

**Comment by luochen1990** [comment]
Yes, I can reproduce this issue, but only for ruby files, I still can't figure out the problem ...
